packages/ckeditor5-ui/src/textarea/textareaview.ts

```diff
@@ -107,12 +124,44 @@ export default class TextareaView extends InputBase<HTMLTextAreaElement> {
 		this.on( 'change:value', () => {
 			// The content needs to be updated by the browser after the value is changed. It takes a few ms.
 			global.window.requestAnimationFrame( () => {
-				if ( isVisible( this.element ) ) {
-					this._updateAutoGrowHeight();
-					this.fire<TextareaViewUpdateEvent>( 'update' );
+				if ( !isVisible( this.element ) ) {
+					this._isUpdateAutoGrowHeightPending = true;
+
+					return;
 				}
+
+				this._updateAutoGrowHeight();
+				this.fire<TextareaViewUpdateEvent>( 'update' );
 			} );
 		} );
+
+		// This logic handles cases when textarea has been hidden while changes were made to its content but they could not
+		// be reflected to its height (auto-grown) because it was invisible or detached from DOM. The auto-grow logic
+		// gets executed as soon as the textarea becomes visible again.
+		this._resizeObserver = new ResizeObserver( this.element!, evt => {
```

Comment:
I tried to figure out why we ended up with this solution and why we need `ResizeObserver` for this. It is used only for the case described in the comment, which made me even more confusing.

In principle, what we want to do is: when the textarea becomes visible, we want to auto-resize it (as it's value may have changed while it was not visible).

But why we need `ResizeObserver` for that? AFAIR -- we don't have a precise way to observe when given element becomes visible/hidden if it is done outside of editor's control. However, I assume that `ResizeObserver` detects that, because when something gets hidden/visible its size becomes 0/non-0.

Is this correct?

If yes, then please change the comment so it describes what we want to achieve in simpler words and why we need `ResizeObserver` for that.

Reply:
For the context - the root cause for the whole change was to simplify the code in AI Assistant (https://github.com/cksource/ckeditor5-commercial/pull/5832#discussion_r1426461711). We didn't want to make people use the additional listener to reset their components before dialog is hidden, and `onHide` effectively works as `afterHide`. Hence we're making the `Textarea` responsible for detecting when it can resize itself.

I added more elaborate comment explaining the generic issue.